Set language in docker image to C.UTF8

Our docker image is set to using a non UTF8 language as its default.  The result of this is that we cannot store files that contain extended characters in their names.  The fix for this is to set the environmental variable `lang` in our docker image to `C.UTF8`.  You can see this work by adding the environmental variable to a docker config:
```
    environment:
      LANG: 'C.UTF-8'
```
We need to add this environmental variable by default to our docker image




To test example create a PDF file called `July groups – net adjustment results (145 KB PDF), updated May 2019.pdf` and try to upload it into a dockerized dotCMS.  This will fail with a message like:

```
[28/09/22 21:04:37:340 UTC] ERROR temp.TempFileResource: Malformed input or input contains unmappable characters: /data/shared/assets/tmp_upload/temp_a547a9e844/July groups Ã¢Â€Â" net adjustment results (145 KB PDF), updated May 2019.pdf
com.dotmarketing.exception.DotRuntimeException: Malformed input or input contains unmappable characters: /data/shared/assets/tmp_upload/temp_a547a9e844/July groups Ã¢Â€Â" net adjustment results (145 KB PDF), updated May 2019.pdf
at com.dotcms.rest.api.v1.temp.TempFileAPI.createTempFile(TempFileAPI.java:185) ~[dotcms_22.03.2_999999.jar:?]
at com.dotcms.rest.api.v1.temp.TempFileResource.lambda$0(TempFileResource.java:171) ~[dotcms_22.03.2_999999.jar:?]
Caused by: java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /data/shared/assets/tmp_upload/temp_a547a9e844/July groups Ã¢Â€Â" net adjustment results (145 KB PDF), updated May 2019.pdf
at com.dotcms.rest.api.v1.temp.TempFileAPI.createTempFile(TempFileAPI.java:172) ~[dotcms_22.03.2_999999.jar:?]
... 7 more
```
 





export LANG=C.UTF8

--- a/cicd/docker/dotcms-compose.yml
+++ b/cicd/docker/dotcms-compose.yml
@@ -22,6 +22,7 @@ services:
       - 8443:8443
       - 8000:8000
     environment:
+      LANG: 'C.UTF-8'
       databaseType: "${DB_TYPE}"
       CATALINA_OPTS: "-XX:+PrintFlagsFinal"
       DB_BASE_URL: "jdbc:postgresql://database/dotcms"

--- a/docker/docker-compose-examples/cluster-mode/docker-compose-node-1.yml
+++ b/docker/docker-compose-examples/cluster-mode/docker-compose-node-1.yml
@@ -29,6 +29,7 @@ services:
   dotcms-node-1:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/cluster-mode/docker-compose-node-2.yml
+++ b/docker/docker-compose-examples/cluster-mode/docker-compose-node-2.yml
@@ -13,6 +13,7 @@ services:
   dotcms-node-2:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/push-publish/docker-compose-receiver.yml
+++ b/docker/docker-compose-examples/push-publish/docker-compose-receiver.yml
@@ -28,6 +28,7 @@ services:
   dotcms-receiver:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db-receiver/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/push-publish/docker-compose-sender.yml
+++ b/docker/docker-compose-examples/push-publish/docker-compose-sender.yml
@@ -28,6 +28,7 @@ services:
   dotcms-sender:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db-sender/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/single-node-debug-mode/docker-compose.yml
+++ b/docker/docker-compose-examples/single-node-debug-mode/docker-compose.yml
@@ -29,6 +29,7 @@ services:
   dotcms:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "CMS_JAVA_OPTS": '-Xmx1g -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=*:8000 '
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'

--- a/docker/docker-compose-examples/single-node-demo-site/docker-compose.yml
+++ b/docker/docker-compose-examples/single-node-demo-site/docker-compose.yml
@@ -29,6 +29,7 @@ services:
   dotcms:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "CATALINA_OPTS": '-Xmx1g '
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'

--- a/docker/docker-compose-examples/single-node/docker-compose.yml
+++ b/docker/docker-compose-examples/single-node/docker-compose.yml
@@ -29,6 +29,7 @@ services:
   dotcms:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "CATALINA_OPTS": '-Xmx1g '
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'

--- a/docker/docker-compose-examples/with-kibana/docker-compose.yml
+++ b/docker/docker-compose-examples/with-kibana/docker-compose.yml
@@ -38,6 +38,7 @@ services:
   dotcms:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/with-mssql-arm/docker-compose.yml
+++ b/docker/docker-compose-examples/with-mssql-arm/docker-compose.yml
@@ -25,6 +25,7 @@ services:
   dotcms-dev:
     image: dotcms/dotcms:latest
     environment:
+      "LANG": 'C.UTF-8'
       "CATALINA_OPTS": " -Xms1g -Xmx1g "
       "DB_BASE_URL": "jdbc:sqlserver://db-dev;databaseName=master"
       "DB_USERNAME": "sa"

--- a/docker/docker-compose-examples/with-mssql/docker-compose.yml
+++ b/docker/docker-compose-examples/with-mssql/docker-compose.yml
@@ -25,6 +25,7 @@ services:
   dotcms-dev:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "CATALINA_OPTS": ' -Xms1g -Xmx1g '
         "DB_BASE_URL": "jdbc:sqlserver://db-dev:1433;databaseName=dotcms"
         "DB_USERNAME": 'SA'

--- a/docker/docker-compose-examples/with-redis/docker-compose-node-1.yml
+++ b/docker/docker-compose-examples/with-redis/docker-compose-node-1.yml
@@ -30,6 +30,7 @@ services:
   dotcms-node-1:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

--- a/docker/docker-compose-examples/with-redis/docker-compose-node-2.yml
+++ b/docker/docker-compose-examples/with-redis/docker-compose-node-2.yml
@@ -15,6 +15,7 @@ services:
   dotcms-node-2:
     image: dotcms/dotcms:latest
     environment:
+        "LANG": 'C.UTF-8'
         "DB_BASE_URL": "jdbc:postgresql://db/dotcms"
         "DB_USERNAME": 'dotcmsdbuser'
         "DB_PASSWORD": 'password'

